install/docker/debian-docker-install.md

```diff
@@ -0,0 +1,121 @@
+# Install Docker for Kata Containers on Debian
+
+> **Note:**
+>
+> - This guide assumes you have
+>   [already installed the Kata Containers packages](../debian-installation-guide.md).
+> - this guide allows for installation with `systemd` or `sysVinit`init systems 
+> - notes are also provided for running multiple runtimes
+
+1. Install the latest version of Docker with the following commands:
+
+   > **Note:** This step is only required if Docker is not installed on the system.
+
+   ```bash
+   $ sudo apt-get install apt-transport-https ca-certificates curl gnupg2 software-properties-common  
+   $ curl -fsSL https://download.docker.com/linux/$(. /etc/os-release; echo "$ID")/gpg | sudo apt-key add -
+   $ sudo add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/$(. /etc/os-release; echo "$ID") $(lsb_release -cs) stable"
+   $ sudo apt-get update
+   $ sudo apt-get install docker-ce
+   ```
+
+   For more information on installing Docker please refer to the
+   [Docker Guide](https://docs.docker.com/engine/installation/linux/debian).
+
+2. Configure Docker to use Kata Containers by default with one of the following methods:
+
+    > **Note:** It is possible to have two docker runtimes enabled and pick the option at config/start of a docker machine.
+    > - This is useful for running Clear Containers and Kata Containers concurrently, for instance.
+    > - see [Docker Docs on runtime options](https://docs.docker.com/engine/reference/commandline/dockerd/#docker-runtime-execution-options)
+
+    1. sysVinit
+    
+        - with sysVinit,  docker config is stored in `/etc/default/docker`, edit the options similar to the following: 
+        
+        ```
+        # single runtime
+        DOCKER_OPTS="-D --add-runtime kata-runtime=/usr/bin/kata-runtime --default-runtime=kata-runtime
+        # multiple runtime options (kata default)
+        #DOCKER_OPTS="-D --add-runtime kata-runtime=/usr/bin/kata-runtime cc-runtime=/usr/bin/cc-runtime --default-runtime=kata-runtime
+        ```
+        > Question: When changing the default runtime, does docker remember which runtime a container was previously created with?
+        > Answer: It looks so (probably stored in container json), also  note the [support for specifying `runtime: `](https://github.com/docker/compose/pull/5405) in docker-compose (v3)
+        
+        > **Note:** other options can be added to the `DOCKER_OPTS` in `/etc/default/docker` either as separate commands or as a unified command,
+        > for instance, the following is valid syntax:
+        
+        ```
+        DOCKER_OPTS="-D --add-runtime kata-runtime=/usr/bin/kata-runtime --default-runtime=kata-runtime \
+        --storage-driver devicemapper \
+        --storage-opt dm.basesize=40G \
+        --storage-opt dm.thinpooldev=/dev/mapper/docker--vg-docker--pool \
+        --storage-opt dm.fs=ext4"
```

Comment:
These three storage options look rather machine-specific to me?

Any thoughts @grahamwhaley, @chavafg?

Reply:
yes, I agree, they are probably superfluous

Reply:
All "in-tree" docs such as these ones go through a relatively rigorous review process (hopefully it's not *too* painful though?! ;)

We try to keep these "official" docs as concise and clear as possible whilst ensuring consistency, accuracy, *etc*. But there is scope I think for a more "agile" set of docs (FAQs, HOWTO's maybe?, *etc*) to supplement the official ones, maybe in:

- https://github.com/kata-containers/community/wiki

That could be were some of this additional info could be stored.

wdyt @grahamwhaley, @egernst, @gnawux?

Reply:
Indeed, they are generic docker options that we don't include in the other distro docker install guides afaict - we should keep this file minimal, like the others in this dir.
We could, and probably should, have somewhere something that makes it clear that the generic docker settings, like those above for a specific graph driver, are still applicable and work with kata, but we can then point off to the docs.docker.com site. We don't have to create a new doc/place along with this PR though. Maybe that is something that would fall into my long-pending 'big book of kata' doc ...